src/Audio.md

```diff
@@ -0,0 +1,103 @@
+# Audio Overview
+
+Game Boy audio is sometimes called "8-bit", but this doesn't mean much: the term is used to designate audio similar to that produced by 8-bit consoles.
+
+A better description would be "chiptune": tunes produced by a chip.
+This section describes how audio is generated on the Game Boy.
+
+## Architecture
+
+{{#include imgs/apu_overview.svg:2:}}
+
+The Game Boy has four sound generation units, called **channels** 1 through 4, notated "CH1", "CH2", etc.
+Unlike other sound chips, such as the SNES' S-DSP or the Atari 5200's POKEY, each sound channel is largely different from the others.
+
+Each channel generates an electronic signal; these signals are then mixed into two new channels (for stereo: one for the left ear, one for the right ear), which are then individually amplified, and then output either to the headphone jack, or the speaker[^speaker_mono].
+
+Channels 1 and 2, the "pulse channels", produce [pulse (rectangular) waves](https://en.wikipedia.org/wiki/Pulse_wave).
+Channel 3, the "wave" channel, produces user-defined waves.
+Channel 4 is the "[noise](https://en.wikipedia.org/wiki/Noise_in_music)" channel, producing a pseudo-random wave.
+
+The VIN channel is a signal received directly from the cartridge, allowing external hardware to supply a fifth sound channel.
+No licensed games used this feature, and it was omitted from the Game Boy Advance.
```

Comment:
Got some links?

Reply:
Nanoloop: https://www.nanoloop.com/mono/index.html

PocketVoice:
http://n64.icequake.net/mirror/www.bung.com.hk/porducts/pocketvoice.htm
http://n64.icequake.net/mirror/www.bung.com.hk/support/download.htm

Nanoloop is still being sold, whereas PocketVoice hasn't been sold in about 20 years since Bung went under, and was probably one of their more obscure products. I think it's enough for now to just mention in passing that "no licensed games used the VIN pin, although two unlicensed cartridges, Bung PocketVoice, and Nanoloop Mono did."

Speaking of which... we shold probably also explain how...

- the VIN pin is sometimes used as a flash /WR pin.
- how it therefore often has a pullup resistor.
- how this means the behavior of the modulating VIN can produce different results depending on the cartridge it's running from.
- How one homebrew game (which I don't remember the name of) mistakenly used 'rNR50` as a 2x4 bit PCM register, because on the cart it was developed on the DC offset on VIN was just right for this to work ok, but won't always work as per the previous point.

Reply:
Mm, honestly, I'd rather not touch VIN. This PR is enough of a brain space nightmare as-is. Open a separate issue, maybe?